[Cookbook][Routing] Update custom_route_loader.rst

xelaris opened the pull request:
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.3+ |
| Fixed tickets |  |

Just some tweaks (including "Acme\DemoBundle" => "AppBundle").

While reading this chapter I wonder, if 

> A custom route loader allows you to add routes to an application without
>  including them, for example, in a YAML file. [...] This may be especially important when you want
>  to make the bundle reusable, or when you have open-sourced it as this would
>  slow down the installation process and make it error-prone.

may raise false expectations. Since also with a route loader it is still required to add something to the routing.yml (or xml). I don't see the benefit of using a custom route loader in a reusable bundle over importing a YAML file from a third-party bundle like FOSUserBundle. It makes totally sense for something like the FOSRestBundle does (generating routes based on conventions), but the chapter introduction sounds, like a Bundle could load its routes without manual tweaks to the config. If I am not wrong, this isn't possible, isn't it?

xelaris committed:
[Cookbook][Routing] Update custom_route_loader.rst

WouterJ commented:
I agree with you, I see no benefit of a custom routing loader.

Some changes in this PR might conflict with #4740. Could you please check that and remove the ones that will conflict?

xelaris commented:
@WouterJ The file `cookbook/routing/custom_route_loader.rst` is not part of #4740, isn't it?

WouterJ commented:
Indeed, my bad. I thought it changed all cookbook articles.

stof commented on cookbook/routing/custom_route_loader.rst:
```diff
@@ -14,13 +14,14 @@ slow down the installation process and make it error-prone.
     accomplish cases like those described above, for instance
-    `FOSRestBundle`_, `JMSI18nRoutingBundle`_, `KnpRadBundle`_ and `SonataAdminBundle`_.
+    `FOSRestBundle`_, `JMSI18nRoutingBundle`_, `KnpRadBundle`_ and
+    `SonataAdminBundle`_.
```
we might want to add AsseticBundle in the list actually

WouterJ replied:
I would rather see a shorter list than making it longer

weaverryan commented:
@xelaris this looks great - nice work! But you're right - the first 2 paragraphs (especially the first) is terrible! I think we should mention FOSRestBundle as we do now, and make a new intro centered around it as one example.

Also, it's out of scope, but I think we should just show the class extending `Loader` in the first example. That's much easier and more powerful. Then we can have a note that says that you can simply implement `LoaderInterface` instead of extending `Loader` if you want.

@xelaris Can you at least take a shot at updating the first paragraph for us?

xabbuh commented:
ping @xelaris :)

xelaris commented:
Thank you for the ping @xabbuh. I will try to go on with this on the weekend.

xelaris committed:
Rewrite first paragraph

xelaris committed:
Change code example to extend from Loader class

xelaris commented:
@weaverryan Thank you for the feedback. I have changed the first two paragraphs and tried to make it more clear.
In addition I have changed the code example to extend from `Loader` as you suggested.

xelaris commented:
ping @weaverryan

xabbuh commented on cookbook/routing/custom_route_loader.rst:
```diff
@@ -107,32 +113,35 @@ type you want. The resource name itself is not actually used in the example::
+have to create an ``extraAction`` method in the ``ExtraController``
+of the ``AppBundle``::
+
+    namespace AppBundle\Controller;
```
you should add the filename comment above this:

```php
// src/AppBundle/Controller/ExtraController.php
```

xelaris committed:
Add filename comments to code blocks

xelaris commented:
Thanks @xabbuh. I have added filename comments to all the php code blocks.

WouterJ commented:
I don't think anything is missing here. Flagged it with Finished :)

xabbuh commented:
I agree :+1:

weaverryan commented:
This looks really awesome - I have no changes. Thanks so much Alexander!
